modules/collect/src/main/java/com/opengamma/strata/collect/MapStream.java

```diff
@@ -185,6 +171,27 @@
     return new MapStream<>(Stream.empty());
   }
 
+  /**
+   * Creates a stream of map entries whose elements are those of the first map stream followed by those of the second.
```

Comment:
According to the underlying `Stream.concat`, the order is only maintained if both inputs are ordered, so you can't promise "followed by" unless both inputs are ordered. 

There isn't an obvious way to force ordering - I guess it would require buffering the input streams - so I guess we just don't worry about the problem.